src/arviz_plots/plots/lm_plot.py

```diff
@@ -0,0 +1,324 @@
+"""lm plot code."""
+from collections.abc import Mapping, Sequence
+from copy import copy
+from importlib import import_module
+from typing import Any, Literal
+
+import numpy as np
+import xarray as xr
+from arviz_base import rcParams
+from scipy.interpolate import griddata
+from scipy.signal import savgol_filter
+
+from arviz_plots.plot_collection import PlotCollection
+from arviz_plots.plots.utils import (
+    filter_aes,
+    get_contrast_colors,
+    process_group_variables_coords,
+    set_wrap_layout,
+)
+from arviz_plots.visuals import fill_between_y, labelled_x, labelled_y, line_xy, scatter_xy
+
+
+def plot_lm(
+    dt,
+    target_data=None,
+    independent_data=None,
+    prediction_data=None,
+    ci_mean_independent_data=None,
+    filter_vars=None,
+    prediction_group="predictions",
+    coords=None,
+    sample_dims=None,
+    ci_prob=None,
+    ci_kind=None,
+    smooth=True,
+    plot_collection=None,
+    backend=None,
+    aes_by_visuals: Mapping[
+        Literal[
+            "line",
+            "mean_line",
+            "fill",
+            "scatter",
+            "xlabel",
+            "ylabel",
+        ],
+        Sequence[str],
+    ] = None,
+    visuals: Mapping[
+        Literal[
+            "line",
+            "mean_line",
+            "fill",
+            "scatter",
+            "xlabel",
+            "ylabel",
+        ],
+        Mapping[str, Any] | Literal[False],
+    ] = None,
+    stats: Mapping[
+        Literal["credible_interval",],
+        Mapping[str, Any] | xr.Dataset,
+    ] = None,
+    **pc_kwargs,
+):
+    """Posterior predictive and mean plots for regression-like data..
+
+    Parameters
+    ----------
+    dt : DataTree
+        Input data
+
+    """
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+    if visuals is None:
+        visuals = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if backend is None:
+        if plot_collection is None:
+            backend = rcParams["plot.backend"]
+        else:
+            backend = plot_collection.backend
```

Comment:
I think this was needed a while ago before we simplified all the automatic figsizing logic. `PlotCollection.grid` and `.wrap` can take `None` as backend and will use the rcParam, so we can simplify this to

```
backend = plot_collection.backend
```

if we put it after we initialize the plot_collection